fastchat/serve/gradio_block_arena_vision_anony.py

```diff
@@ -242,6 +246,71 @@ def clear_history(request: gr.Request):
         + [""]
     )
 
+def extract_text_from_pdf(pdf_file_path):
+    """Extract text from a PDF file."""
+    try:
+        with open(pdf_file_path, 'rb') as f:
+            reader = PyPDF2.PdfReader(f)
+            pdf_text = ""
+            for page in reader.pages:
+                pdf_text += page.extract_text()
+            return pdf_text
+    except Exception as e:
+        logger.error(f"Failed to extract text from PDF: {e}")
+        return None
+    
+import os
+import nest_asyncio
+from llama_parse import LlamaParse
+
+nest_asyncio.apply()  # Ensure compatibility with async environments
+
+def pdf_parse(pdf_path):
+    # Set API key, can also be configured in the environment
+    api_key = "LLAMA API"
+
+    # Initialize the LlamaParse object
+    parser = LlamaParse(
+        api_key=api_key, 
+        result_type="markdown",  # Output in Markdown format
+        num_workers=4,           # Number of API calls for batch processing
+        verbose=True,            # Print detailed logs
+        language="en"            # Set language (default is English)
```

Comment:
what if the pdf is in different language?

Reply:
This is optional, we can remove this

Reply:
I don't know if LlamaParse can handle identifying language. Maybe we have to use language detection or translator?